scripts/up.sh

```diff
@@ -44,15 +65,23 @@ coredns_image="multicluster/coredns:latest"
 coredns_path="/tmp/coredns-${coredns_version}"
 if [ ! -d "${coredns_path}" ]; then
   pushd /tmp
-  curl -fLO "https://github.com/coredns/coredns/archive/refs/tags/v${coredns_version}.tar.gz"
-  tar xf v${coredns_version}.tar.gz
+  # Since the coredns make command uses the git command, we clone it with git
+  git clone --depth 1 https://github.com/coredns/coredns.git --branch v${coredns_version} --single-branch "${coredns_path}"
+
+  # coredns binary is not multi-arch, so we need to download the binary for the current platform
+  platform="$(detect_platform)"
+  coredns_binary_url="https://github.com/coredns/coredns/releases/download/v${coredns_version}/coredns_${coredns_version}_${platform}.tgz"
+  curl -fLO "${coredns_binary_url}"
+  tar xzf "coredns_${coredns_version}_${platform}.tgz"
+  chmod +x coredns
```

Comment:
The script rebuild the coredns binary because we need to add an out of tree plugin (just a few lines below) so we cannot use the prebuild binary unfortunately

Reply:
Thank you for your comment.

I reviewed the current `Dockerfile` ([link](https://github.com/coredns/coredns/blob/master/Dockerfile#L14)) and noticed that it primarily utilizes the `docker build` command, which copies the existing CoreDNS binary rather than rebuilding it. This suggests that the binary is not being reconstructed within this specific Docker build process.

Could you please provide more details or point me to the specific script or step where the CoreDNS binary is being rebuilt? Understanding the exact location or method used for the rebuild would greatly help in addressing the integration of the out-of-tree plugin.

I appreciate your guidance and assistance on this matter.

Reply:
Sure that's when `make` is ran right before the `docker build` command: https://github.com/kubernetes-sigs/mcs-api/blob/fede3192824f8c9d44719a467528f9438ae6007b/scripts/up.sh#L55

Reply:
I see. Is there a good way to get binaries built that match the container execution environment, rather than binaries that match the host OS environment?
MacOS is hosted on Darwin, but the execution environment is linux.

Reply:
Hmm I see :thinking:, maybe something similar to this then: https://github.com/coredns/coredns?tab=readme-ov-file#compilation-with-docker?

Reply:
Thx! 🙏